Code/Common/src/sitkCreateInterpolator.hxx

```diff
@@ -0,0 +1,269 @@
+  return nullptr;
+}
+
+template <typename TPixelType, unsigned int N>
+typename itk::InterpolateImageFunction<itk::VectorImage<TPixelType, N>,
+                                       double>::Pointer
+CreateInterpolator(const itk::VectorImage<TPixelType, N> * image,
+                   InterpolatorEnum itype) {
+  Unused(image);
+  using TImageType = itk::VectorImage<TPixelType, N>;
+  using RType =
+      typename itk::InterpolateImageFunction<TImageType, double>::Pointer;
+  using BoundaryCondition =
+      typename itk::ZeroFluxNeumannBoundaryCondition<TImageType, TImageType>;
+
+  switch (itype) {
+  case sitkNearestNeighbor: {
+    using InterpolatorType =
+        itk::NearestNeighborInterpolateImageFunction<TImageType, double>;
+    return RType(InterpolatorType::New());
+  }
+  case sitkLinear: {
+    using InterpolatorType =
+        itk::LinearInterpolateImageFunction<TImageType, double>;
+    return RType(InterpolatorType::New());
+  }
+  default:
+    return nullptr;
+  }
+}
+
+
+template <typename TPixelType, unsigned int N>
+typename itk::InterpolateImageFunction<itk::Image<std::complex<TPixelType>, N>,
+                                       double>::Pointer
+CreateInterpolator(const itk::Image<std::complex<TPixelType>, N> * image,
+                   InterpolatorEnum itype) {
+  Unused(image);
+  using TImageType = itk::Image<std::complex<TPixelType>, N>;
+  using RType =
+      typename itk::InterpolateImageFunction<TImageType, double>::Pointer;
+  using BoundaryCondition =
+      typename itk::ZeroFluxNeumannBoundaryCondition<TImageType, TImageType>;
+
+  switch (itype) {
+  case sitkNearestNeighbor: {
+    using InterpolatorType =
+        itk::NearestNeighborInterpolateImageFunction<TImageType, double>;
+    return RType(InterpolatorType::New());
+  }
+  case sitkLinear: {
+    using InterpolatorType =
+        itk::LinearInterpolateImageFunction<TImageType, double>;
+    return RType(InterpolatorType::New());
+  }
+  default:
+    return nullptr;
+  }
+}
+
+template <typename TImageType>
+typename itk::InterpolateImageFunction<TImageType, double>::Pointer
+CreateInterpolator(const TImageType *image, InterpolatorEnum itype) {
+  using RType =
+      typename itk::InterpolateImageFunction<TImageType, double>::Pointer;
+  using BoundaryCondition =
+      typename itk::ZeroFluxNeumannBoundaryCondition<TImageType, TImageType>;
+
+  using SpacingType = typename TImageType::SpacingType;
+
+  static const unsigned int WindowingRadius = 5;
+
+  const SpacingType &spacing = image->GetSpacing();
+
+  switch (itype) {
+  case sitkNearestNeighbor: {
+    using InterpolatorType =
+        itk::NearestNeighborInterpolateImageFunction<TImageType, double>;
+    return RType(InterpolatorType::New());
+  }
+  case sitkLinear: {
+    using InterpolatorType =
+        itk::LinearInterpolateImageFunction<TImageType, double>;
+    return RType(InterpolatorType::New());
+  }
+  case sitkBSpline: {
```

Comment:
Explicitly set BSpline order to 3 or add documentation stating that this is the default order.

Reply:
That is documented in sitkInterpoaltor.h.